### DISCUSSION TITLE
An extension to enable offline Jupyter Books

### DISCUSSION THREAD

**Discussion by Daltz333** [executablebooks/meta#691] [Show and tell]
Hiya folks. My name is Dalton Smith and I'm the documentation maintainer for [WPILib](https://github.com/wpilibsuite/frc-docs). I'm responsible for the creating/publishing/maintaining of various sphinx extensions such as

- [sphinxext-opengraph](https://github.com/wpilibsuite/sphinxext-opengraph)
- [sphinxext-rediraffe](https://github.com/wpilibsuite/sphinxext-rediraffe)
- [sphinxext-presentations](https://github.com/wpilibsuite/sphinxext-presentations)
- [sphinxext-photofinish](https://github.com/wpilibsuite/sphinxext-photofinish)
- [sphinxext-delta](https://github.com/wpilibsuite/sphinxext-delta)
- [sphinxext-mimic](https://github.com/wpilibsuite/sphinxext-mimic)
- [sphinxext-remoteliteralinclude](https://github.com/wpilibsuite/sphinxext-remoteliteralinclude)
- [sphinxext-toptranslators](https://github.com/wpilibsuite/sphinxext-toptranslators)

Anyways, I'm not developing an extension that uses [Progressive Web Application](https://web.dev/what-are-pwas/) to turn Sphinx websites into offline capable web-applications complete with push notification support. I've been a following of the Juypterbook project for awhile, and am partially involved with [sphinx-tabs](https://github.com/executablebooks/sphinx-tabs). 

I am wondering if anyone familiar with Juypterbook is able to trial out [sphinxext-pwa](https://github.com/wpilibsuite/sphinxext-pwa) and give me some feedback and information. I specifically singled out Juypterbook as it is one of the most complex and full-fledged Sphinx projects out there!

The only requirements is that you add the PWA to your `conf.py` or what the equivalent is for Juypterbook (currently not deployed to PyPI), *but* you do also need Node>=16 for deployment on ReadTheDocs (see the README of [sphinxext-pwa](https://github.com/wpilibsuite/sphinxext-pwa))

Last other node, iOS support is sketchy because they have a ton of open webkit related bugs regarding PWAs.

Feel free to reply to this discussion about any feedback, and sorry if this is not the appropriate place to put this!

Sincerely,
Dalton Smith

**Comment by choldgraf** [comment]
Just want to note that this seems really cool! I don't have a ton of bandwidth right now but would love if somebody wanted to give this a shot. @Daltz333 what are the kinds of feedback you're looking for? What are the things a person should do when trying out your extension? (e.g. should they try to add it to their mobile homescreen or something?)

**Reply by Daltz333** [reply]
Yeah, specifically I'm looking for usage of this extension in Juypterbook projects. They should install the extension with the instructions listed above (simple I swear!) and then add the app to their homescreen.
